### PR TITLE
Support all configuration options for initial position when processing events using EventProcessorClient

### LINKED ISSUE #26434: [FEATURE REQ] Support all configurations for the initial position for an Event Hub partitition
**Is your feature request related to a problem? Please describe.**
In the 4.0.0-beta.2 only `earliest` and `latest` positions are supported for configuring the consuming initial position, but the `EventPostition` in EventProcessorClient supports passing offset, sequence number, and enquedDatetime. We should support all of them like this:

```yaml
spring:
  cloud:
    stream:
      eventhubs:
        bindings:
          consume-in-0:
            consumer:
              initial-partition-event-position:
                0:
                  offset: earliest
                1:
                  sequence-number: 100
                2:
                  enqueued-date-time: 2022-01-12T13:32:47.650005Z
                3: 
                  offset: 19400
```

**Describe the solution you'd like**
Support all configuration options for the starting position.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

### PR DESCRIPTION
**Pull request by saragluna**
To resolve #26434.

### CODE REVIEW

**Comment by saragluna** [comment]
still need to improve the configuration properties hint.

**Comment by chenrujun** [comment]
> still need to improve the configuration properties hint.

How about convert it to draft?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/properties/AzureEventHubsProperties.java [hunk 1]
```diff
@@ -228,6 +229,65 @@ public BlobCheckpointStore getCheckpointStore() {
...
+             * The offset of the event within that partition. String keyword, "earliest" and "latest" (case-insensitive),
+             * are reserved for specifying the start and end of the partition. Other provided value will be cast to Long.
+             */
+            private String offset;
+            /**
+             * The sequence number of the event within that partition.
+             */
+            private Long sequenceNumber;
+            /**
+             * The event enqueued after the requested enqueuedDateTime becomes the current position.
```
Comment (yiliuTo) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/properties/AzureEventHubsProperties.java:247]: This member is an Instant, thus is the javadoc should be "The event enqueued time after the requested enqueuedDateTime becomes the current position."

File: sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/service/eventhubs/properties/EventProcessorClientProperties.java [hunk 2]
```diff
@@ -69,17 +69,33 @@
...
+        String getOffset();
+
+        /**
+         * The sequence number of the event within that partition.
+         * @return The sequence number of the event within that partition.
+         */
+        Long getSequenceNumber();
+
+        /**
+         * The event enqueued after the requested enqueuedDateTime becomes the current position.
```
Comment (yiliuTo) [inline sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/service/eventhubs/properties/EventProcessorClientProperties.java:94]: same here

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**